pkg/apiserver/handlers.go

```diff
@@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	done := make(chan struct{}, 1)
+	done := make(chan struct{})
 	tw := newTimeoutWriter(w)
 	go func() {
 		t.handler.ServeHTTP(tw, r)
```

Comment:
This is the crash path. Panic in a goroutine bringing down the API server. In addition to inverting the timeout and recover handlers in genericapiserver.go, it wouldn't hurt to add a recover here as well just to be safe.